lib/martian_robot.rb

```diff
@@ -28,4 +33,14 @@ def starting_position
     [x,y]
   end
 
+  private
+
+  def is_robot_lost?(final_position)
+    is_invalid_coordinate?(final_position[0]) || is_invalid_coordinate?(final_position[1])
```

Comment:
Depending on the content of final_position, you may have a _more_ Ruby to write it with:

``` ruby
final_posititions.all? { |position| is_invalid_coordinate?(position) }
```

https://ruby-doc.org/core-2.4.1/Enumerable.html